delira/training/experiment.py

```diff
@@ -118,20 +132,48 @@ def kfold(self, num_epochs: int, data: typing.List[BaseDataManager],
 
         """
 
+        # set number of splits if not specified
         if num_splits is None:
-            num_splits = len(data)
-
+            if isinstance(data, BaseDataManager):
+                num_splits = len(data.dataset)
```

Comment:
I don't think setting num_splits = len(dataset) is a good idea. If for example we have 5000 training samples we will get 5000 splits. Better throw a warning and reduce it to some commonly used number like 5 or at most 10.

Reply:
Changed default value to 10 splits